airbyte-local-cli-nodejs/src/utils.ts

```diff
@@ -37,13 +38,65 @@ export async function parseConfigFile(configFilePath: string) {
   }
 }
 
+// Run a command and throw an error if it fails
+function execCommand(command: string, args: string[], options?: {errMsg: string}) {
+  const result = spawnSync(command, args, {shell: false});
+
+  if (result.error || result.status !== 0) {
+    const errInfoMsg = options?.errMsg ? `${options?.errMsg}: ` : '';
+    const errMsg = (result.error ? result.error?.message : result.stderr?.toString()) ?? `unknown error`;
+    throw new Error(`${errInfoMsg}${errMsg}`);
+  }
+}
+
 // Check if Docker is installed
 export function checkDockerInstalled(command = 'docker', args = ['--version']) {
-  const result = spawnSync(command, args, {shell: false});
+  execCommand(command, args, {errMsg: 'Docker is not installed'});
+}
+
+// Create a temporary directory
+export function createTmpDir(tmpDir?: string) {
+  logger.debug(`Creating temporary directory for temporary Airbyte files...`);
+  const currentDir = process.cwd();
+  const tmpDirPath = mkdtempSync(tmpDir ?? `${currentDir}${sep}tmp-`);
```

Comment:
I think a better default is a new directory in the OS temp directory, which you can get from `os.tmpdir()`